### PR TITLE
Naruto run fix

### PR DESCRIPTION
**Pull request by skaljac**
This PR is solving 2 issues:
https://github.com/webaverse/app/issues/1938
https://github.com/webaverse/app/issues/1918

Steps for testing:
Case 1
Press C for crouch
Activate Naruto run with Shift and then double press on button W

* Naruto run should not be possible from crouch mode

Case 2:
Press and hold shift key
double press W or A or S or D and Naruto run should be activated

Double press on Shift key should not activate Naruto run

### CODE REVIEW

File: io-manager.js [hunk 1]
```diff
@@ -64,8 +64,8 @@ ioManager.keys = {
   space: false,
   ctrl: false,
 };
-let lastShiftDownTime = 0;
-ioManager.getLastShiftDownTime = () => lastShiftDownTime;
+let lastWASDDownTime = [0,0,0,0];
```
Comment (avaer) [inline io-manager.js:67]: I would label this with keys, not mysterious indices.

Comment (skaljac) [reply io-manager.js:67]: Changing that now

**Review by avaer — APPROVED** [review]

**Comment by amnaarshed4224** [comment]
It seems to work fine. A few clarifications:
-Holding +double tap WASD activates the Naruto run/fly/travel(Dragon, hovercraft). Once activated, we don't need to hold WASD key, just holding shift keeps avatar in Naruto mode. 
-While running when lets say shift+W is being already pressed, double tapping any other key A,S,D activates Naruto mode

Are these intentional?

**Comment by avaer** [comment]
@skaljac ☝️

**Comment by skaljac** [comment]
@amnaarshed4224 @avaer  From original issue it is not clear how exactly this should work. I taught that we should be able to change direction while we are in naruto run mode. This is only possible if we break naruto mode run only when shift key is up. I can easily change this behaviour if we need that. Please let me know what to do?

File: io-manager.js [hunk 2]
```diff
@@ -296,13 +302,29 @@ ioManager.keydown = e => {
       if (!document.pointerLockElement) {
         game.menuVertical(-1);
       }
+
+      const now = Date.now();
```
Comment (avaer) [inline io-manager.js:306]: Should use `performance.now()`, which is what we use elsewhere.

File: io-manager.js [hunk 3]
```diff
@@ -296,13 +302,29 @@ ioManager.keydown = e => {
...
     }
     case 65: { // A
       ioManager.keys.left = true;
       if (!document.pointerLockElement) {
         game.menuHorizontal(-1);
       }
+
+      const now = Date.now();
+      const timeDiff = now - lastWASDDownTime.keyA;
+      if (timeDiff < 200 && ioManager.keys.shift) {
```
Comment (avaer) [inline io-manager.js:323]: `200` should be a constant.

**Pull request merged**